streams/src/main/java/org/apache/kafka/streams/processor/internals/AbstractTask.java

```diff
@@ -190,10 +145,52 @@ public String toString(final String indent) {
         return sb.toString();
     }
 
+    protected Map<TopicPartition, Long> recordCollectorOffsets() {
+        return Collections.emptyMap();
+    }
+
+    protected void initializeOffsetLimits() {
+        for (final TopicPartition partition : partitions) {
+            try {
+                final OffsetAndMetadata metadata = consumer.committed(partition); // TODO: batch API?
+                stateMgr.putOffsetLimit(partition, metadata != null ? metadata.offset() : 0L);
+            } catch (final AuthorizationException e) {
+                throw new ProcessorStateException(String.format("task [%s] AuthorizationException when initializing offsets for %s", id, partition), e);
+            } catch (final WakeupException e) {
+                throw e;
+            } catch (final KafkaException e) {
+                throw new ProcessorStateException(String.format("task [%s] Failed to initialize offsets for %s", id, partition), e);
+            }
+        }
+    }
+
     /**
      * Flush all state stores owned by this task
      */
-    public void flushState() {
+    protected void flushState() {
         stateMgr.flush();
     }
+
+    void initializeStateStores() {
```

Comment:
"package private" is stronger than `protected` and there is no reason to make it more open than required. But I am open to make it `protected`, too. Just don't see why we would need it?